core/src/main/scala/org/apache/spark/internal/config/Python.scala

```diff
@@ -50,4 +50,9 @@ private[spark] object Python {
     .version("2.4.0")
     .bytesConf(ByteUnit.MiB)
     .createOptional
+
+  val PYTHON_GATEWAY_CONNECT_TIMEOUT = ConfigBuilder("spark.python.gateway.connectTimeout")
```

Comment:
Not sure this is the right naming but can't really come up with any better.

Reply:
I think we can just name it like `spark.authenticate.python.socketTimeout` or `spark.python.authenticate.socketTimeout` to look after other configurations `spark.authenticate.*`. In worker side, Py4J gateway is not created.

Reply:
> In worker side, Py4J gateway is not created.

Yeah, that's the reason why I felt it's not right. I would add the second (`spark.python.authenticate.socketTimeout`).